app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args, **kwargs) -> any:
+        cache_key = (args, tuple(sorted(kwargs.items())))
```

Comment:
Consider using `frozenset` instead of `tuple(sorted(kwargs.items()))` for the cache key. This will make the key creation more efficient and avoid unnecessary sorting.